.github/workflows/cicd.yaml

```diff
@@ -160,28 +110,14 @@ jobs:
         with:
           python-version: '3.7'
 
-      - name: get distribution wheels
-        uses: actions/download-artifact@v2
-        with:
-          name: pypi-build-arts
-          path: dist
-
-      - name: list all pypi build arts
-        run: |
-          ls -la dist/
-
-      - name: install check dependencies
+      - name: Install base dependencies
         run: |
-          python -m pip install --upgrade twine
+          python -m pip install --upgrade pip
+          python -m pip install poetry twine
 
-      - name: check pypi artifacts
+      - name: Build and publish
+        env:
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
         run: |
-          twine check dist/*
-
-      - name: publish package to pypi
-        uses: pypa/gh-action-pypi-publish@release/v1
-        with:
-          user: __token__
-          password: ${{ secrets.PYPI_API_TOKEN }}
-          verbose: true
-          skip_existing: true
+          poetry publish --build --username $TWINE_USERNAME --password $TWINE_PASSWORD
```

Comment:
For poetry, `publish` sub-command builds package and uploads to PyPI with `py3-none-any` wheel.

Reply:
For example, this [repo](https://github.com/m3dev/gokart) (I'm a mentainer of this) use poetry publish in [CI](https://github.com/m3dev/gokart/blob/master/.github/workflows/publish.yml#L27), then PyPI has [this files](https://pypi.org/project/gokart/#files)